File: pyround1/504.七进制数.py
#
# @lc app=leetcode.cn id=504 lang=python
#
# [504] 七进制数
#

# @lc code=start
class Solution(object):
    def convertToBase7(self, num):
        """
        :type num: int
        :rtype: str
        """
        if num < 0:
            flag = 1
            num = -num
        else:
            flag = 0
        res = []
        while num>=7:
            fig = str(num % 7) 
            res.append(fig)
            num //= 7
        res.append(str(num))
        if flag:
            res.append('-')
        res.reverse()
        return ''.join(res)

        
# @lc code=end

File: 504.七进制数.cpp
/*
 * @lc app=leetcode.cn id=504 lang=cpp
 *
 * [504] 七进制数
 */

// @lc code=start
class Solution {
public:
    string convertToBase7(int num) {
        if(abs(num) < 7)
            return to_string(num);
        bool flag = false;
        if(num < 0)
        {
            flag = true;
            num = -num;
        }
        string res = "";
        int count = 0;
        while(num){
            int pointer = pow(7, count+1);
            if(num / pointer >= pointer){
                res += to_string(num%pointer);
                num -= num % pointer;
                count++;
            }
            else{
                res += to_string(num/pointer);
                break;
            }
        }
        if(flag){
            res += "-";
        }
        reverse(res.begin(), res.end());
        return res;

    }
};
// @lc code=end

